src/Kinesis.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Cascade\StatusStream;
+
+use Aws\Kinesis\KinesisClient;
+
+class Kinesis implements StatusStreamInterface
+{
+    private $component = '';
+    private $kinesis;
+    private $machineId;
+    private $streamName;
+    private $system = '';
+    private $subsystem = '';
+
+    public function __construct(KinesisClient $kinesis, $streamName)
+    {
+        $this->kinesis = $kinesis;
+        $this->streamName = $streamName;
+    }
+
+    /**
+     * @param string $system
+     * @param string $subsystem
+     * @param string $component
+     */
+    public function setSystemId($system, $subsystem = '', $component = '')
+    {
+        $this->system = $system;
+        $this->subsystem = $subsystem;
+        $this->component = $component;
+    }
+
+    /**
+     * @param array $machineId
+     */
+    public function setMachineId(array $machineId)
```

Comment:
It is at the moment. I'm not sure if I like that or not. It's useful to have it as an array when rendering state details on something like a dashboard, since it makes it easy to group things by the segments of the machine ID -- it might be better called `processID` -- I'll ponder that and might rename.